src/ILLink.Shared/TypeSystemProxy/ParameterCollection.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace ILLink.Shared.TypeSystemProxy
+{
+	internal struct ParameterCollection
```

Comment:
Couple of questions:
* Do we need this? Why not just `IEnumerable<ParameterProxy>` - perf? - if so please add a comment
* Naming - probably better name would be `ParameterProxyCollection`. Or even better `ParameterProxyEnumerable` - or something like that - it's not a Collection really.
* I find it really weird that it doesn't implement `IEnumerable<ParameterProxy>` - even though it would never be used - since compiler will optimize and use the exact types directly, it's really weird. It would also act as a documentation what this is for.

Reply:
Yeah, it was for perf. I like those names better, and I'll add the IEnumerable implementation.